### PR TITLE
Added SSPRK methods

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [X] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Solves a part of #2177

### CODE REVIEW

File: lib/OrdinaryDiffEqSSPRK/Project.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name = "OrdinaryDiffEqSSPRK"
+uuid = "669c94d9-1f4b-4b64-b377-1aa079aa2388"
+authors = ["ParamThakkar123 <paramthakkar864@gmail.com>"]
+version = "1.0.0"
+
+[deps]
+FastBroadcast = "7034ab61-46d4-4ed7-9d0f-46aef9175898"
+MuladdMacro = "46d2c3a1-f734-5fdb-9937-b9b9aeba4221"
+LinearSolve = "7ed4a6bd-45f5-4d41-b270-4a48e9bafcae"
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/Project.toml:9]: What SSPRK method is using a linear solver?

File: test/runtests.jl [hunk 2]
```diff
@@ -35,12 +35,24 @@ function activate_stabilized_rk()
...
 function activate_low_storage_rk()
     Pkg.activate("../lib/OrdinaryDiffEqLowStorageRK")
     Pkg.develop(PackageSpec(path = dirname(@__DIR__)))
     Pkg.instantiate()
 end
 
+function activate_low_storage_rk()
+    Pkg.activate("../lib/OrdinaryDiffEqSSPRK")
+    Pkg.develop(PackageSpec(path = dirname(@__DIR__)))
+    Pkg.instantiate()
```
Comment (ChrisRackauckas) [inline test/runtests.jl:53]: function names

File: test/runtests.jl [hunk 3]
```diff
@@ -164,7 +176,7 @@ end
     end
 
     if !is_APPVEYOR && GROUP == "AlgConvergence_II"
-        @time @safetestset "SSPRK Tests" include("algconvergence/ode_ssprk_tests.jl")
+        @time @safetestset "SSPRK Tests" include("../lib/OrdinaryDiffEqSSPRK/test/ode_ssprk_tests.jl")
```
Comment (ChrisRackauckas) [inline test/runtests.jl:179]: split groups

Comment (ParamThakkar123) [reply test/runtests.jl:179]: image

This way ??

Comment (ChrisRackauckas) [reply test/runtests.jl:179]: Yes, I think I forgot to check that on the last PR too

Comment (ParamThakkar123) [reply test/runtests.jl:179]: I didn't split groups in the last PR. I will do it in this one

**Comment by ParamThakkar123** [comment]
There are some version errors here related to the packages

**Comment by ParamThakkar123** [comment]
image

I ran the runtests.jl inside OrdinaryDiffEqSSPRK and it worked.

**Comment by ChrisRackauckas** [comment]
It's related to https://github.com/YingboMa/FastBroadcast.jl/pull/64. Should be cleaned up in a little bit. Just branch for another set. Maybe do the Feagin next, those should be rather easy.

File: lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl [hunk 4]
```diff
@@ -0,0 +1,25 @@
+module OrdinaryDiffEqSSPRK
+
+import OrdinaryDiffEq: alg_order, alg_adaptive_order, calculate_residuals!,
+                       beta2_default, beta1_default, gamma_default,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, ssp_coefficient,
+                       OrdinaryDiffEqAlgorithm, ispredictive,
+                       OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+                       OrdinaryDiffEqAdaptiveAlgorithm, uses_uprev,
+                       default_controller, PIDController,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl:10]:
```suggestion
```

Clean up what's unused here, seems like it's just a copy paste with extra things carried in.

File: lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl [hunk 5]
```diff
@@ -0,0 +1,25 @@
+module OrdinaryDiffEqSSPRK
+
+import OrdinaryDiffEq: alg_order, alg_adaptive_order, calculate_residuals!,
+                       beta2_default, beta1_default, gamma_default,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, ssp_coefficient,
+                       OrdinaryDiffEqAlgorithm, ispredictive,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl:7]:
```suggestion
                       OrdinaryDiffEqAlgorithm,
```

File: lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl [hunk 6]
```diff
@@ -0,0 +1,25 @@
+module OrdinaryDiffEqSSPRK
+
+import OrdinaryDiffEq: alg_order, alg_adaptive_order, calculate_residuals!,
+                       beta2_default, beta1_default, gamma_default,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl:4]:
```suggestion
```

File: lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl [hunk 7]
```diff
@@ -0,0 +1,25 @@
+module OrdinaryDiffEqSSPRK
+
+import OrdinaryDiffEq: alg_order, alg_adaptive_order, calculate_residuals!,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl:3]:
```suggestion
import OrdinaryDiffEq: alg_order, calculate_residuals!,
```

File: lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+                       beta2_default, beta1_default, gamma_default,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, ssp_coefficient,
+                       OrdinaryDiffEqAlgorithm, ispredictive,
+                       OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+                       OrdinaryDiffEqAdaptiveAlgorithm, uses_uprev,
+                       default_controller, PIDController,
+                       alg_cache, _vec, _reshape, @cache, isfsal, full_cache,
+                       constvalue, _unwrap_val, du_alias_or_new, ArrayFuse
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl:11]:
```suggestion
                       constvalue, _unwrap_val, du_alias_or_new
```

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I started a new PR on #2267 for Feagin

**Comment by ParamThakkar123** [comment]
But it also has the same version errors

**Comment by ChrisRackauckas** [comment]
rebase to latest master

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas Should I close both these PRs and start over again. Because I think there might be some problem on my end.

**Comment by ChrisRackauckas** [comment]
no just. rebase to get the compat fix

**Comment by ChrisRackauckas** [comment]
You just need to move these functions to the lib.

https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_interface.jl#L118-L121

**Pull request merged**